orderapi/src/main/java/fi/hel/verkkokauppa/order/model/Order.java

```diff
@@ -89,4 +92,16 @@ public Order(String orderId, String namespace, String user, LocalDateTime create
         this.createdAt = createdAt;
         this.incrementId = incrementId;
     }
+
+    public Order(String orderId, String namespace, String user, LocalDateTime createdAt, Long incrementId, LocalDateTime lastValidPurchaseDateTime) {
```

Comment:
Why did you create new one? Should you do the parsing for date before and add this value with setter before saving?

Reply:
When aiming to get things working it did not cross my mind to just use setters after initialising the order with the old constructor. But yes, that makes much more sense so I'll remove my constructor and use setters to fill in the lastValidPurchaseDateTime for the order. 👍